src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -246,6 +261,138 @@ export class InstructorCoursesPageComponent implements OnInit {
+    });
+  }
+
+  /**
+   * Creates a new course with the selected feedback sessions
+   */
+  createCourse(result: CopyCourseModalResult): void {
+    this.isLoading = true;
+    this.courseService.createCourse({
+      courseName: result.newCourseName,
+      timeZone: result.newTimeZone,
+      courseId: result.newCourseId,
+    }).subscribe(() => {
+      this.statusMessageService.showSuccessToast('The course has been added.');
+      this.courseService
+        .getAllCoursesAsInstructor('active')
+        .subscribe((resp: Courses) => {
+          const course: Course = resp.courses.find((c: Course) => c.courseId === result.newCourseId) as Course;
+          this.activeCourses.push(this.getCourseModelFromCourse(course));
+          this.activeCoursesDefaultSort();
+          result.chosenFeedbackSessionList.forEach((session: FeedbackSession) => {
+            this.copyFeedbackSession(session, session.feedbackSessionName, result.newCourseId).subscribe();
+          });
+          this.isLoading = false;
+        });
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorToast(resp.error.message);
+      this.isLoading = false;
+      this.hasLoadingFailed = true;
+    });
+  }
+
+  /**
+   * Gets a CourseModel from courseID
+   */
+  private getCourseModelFromCourse(course: Course): CourseModel {
+    let canModifyCourse: boolean = false;
+    let canModifyStudent: boolean = false;
+    if (course.privileges) {
+      canModifyCourse = course.privileges.canModifyCourse;
+      canModifyStudent = course.privileges.canModifyStudent;
+    }
+    const isLoadingCourseStats: boolean = false;
+    const courseModel: CourseModel = Object.assign({},
+        { course, canModifyCourse, canModifyStudent, isLoadingCourseStats });
+    return courseModel;
+  }
+
+  /**
+   * Copies a feedback session.
+   */
+  private copyFeedbackSession(fromFeedbackSession: FeedbackSession, newSessionName: string, newCourseId: string):
+      Observable<FeedbackSession> {
+    let createdFeedbackSession!: FeedbackSession;
+    return this.feedbackSessionsService.createFeedbackSession(newCourseId, {
+      feedbackSessionName: newSessionName,
+      instructions: fromFeedbackSession.instructions,
+
+      submissionStartTimestamp: fromFeedbackSession.submissionStartTimestamp,
+      submissionEndTimestamp: fromFeedbackSession.submissionEndTimestamp,
+      gracePeriod: fromFeedbackSession.gracePeriod,
+
+      sessionVisibleSetting: fromFeedbackSession.sessionVisibleSetting,
+      customSessionVisibleTimestamp: fromFeedbackSession.customSessionVisibleTimestamp,
+
+      responseVisibleSetting: fromFeedbackSession.responseVisibleSetting,
+      customResponseVisibleTimestamp: fromFeedbackSession.customResponseVisibleTimestamp,
+
+      isClosingEmailEnabled: fromFeedbackSession.isClosingEmailEnabled,
+      isPublishedEmailEnabled: fromFeedbackSession.isPublishedEmailEnabled,
```

Comment:
Discussion: is there a way to replace this with something like `feedbackSession.toCreationReqWithName(newSessionName)`?